README.md

```diff
@@ -1,37 +1,78 @@
-* You must submit a pull request to this repo with your code by 9am Monday week
+This is a bowling scorecard calculator. The user inputs an array of scores for each 'frame' (of which there should be 10 in total), each of which is a sub-array with the length of 1, 2 or 3 depending on the score / frame number. The output is the total score.
 
-## The Task
+The calculator will handle bonuses from strikes and spares, plus the rules around the 10th frame. It will also calculate a total score for an incomplete game without throwing errors. Scroll down to see the rules of bowling.
 
-**THIS IS NOT A BOWLING GAME, IT IS A BOWLING SCORECARD PROGRAM. DO NOT GENERATE RANDOM ROLLS. THE USER INPUTS THE ROLLS.**
+To complete this challenge, I followed TDD principles to guide me to my final result. I started with a single method as this was adequate to handle my initial test cases (i.e a gutter game), but as the cases became more complex, I felt it was necessary to move to a single class to take advantage of initialization and private methods. 
 
-Count and sum the scores of a bowling game for one player. For this challenge, you do _not_ need to build a web app with a UI, instead, just focus on the logic for bowling (you also don't need a database). Next end-of-unit challenge, you will have the chance to translate the logic to Javascript and build a user interface.
+I could have refactored my tests to have fewer in the final version, but decided to leave all tests in place to clearly show the TDD process that I followed.
 
-A bowling game consists of 10 frames in which the player tries to knock down the 10 pins. In every frame the player can roll one or two times. The actual number depends on strikes and spares. The score of a frame is the number of knocked down pins plus bonuses for strikes and spares. After every frame the 10 pins are reset.
+I have assumed that the input will always be an array of arrays containing integers, as this would typically be handled at the point of user input (which was not required for this challenge)
 
-As usual please start by
 
-* Forking this repo
 
-* Finally submit a pull request before Monday week at 9am with your solution or partial solution.  However much or little amount of code you wrote please please please submit a pull request before Monday week at 9am. 
+## Planning - handling the rules
 
-___STRONG HINT, IGNORE AT YOUR PERIL:___ Bowling is a deceptively complex game. Careful thought and thorough diagramming — both before and throughout — will save you literal hours of your life.
 
-## Focus for this challenge
-The focus for this challenge is to write high-quality code.
 
-In order to do this, you may pay particular attention to the following:
-* Using diagramming to plan your approach to the challenge
-* TDD your code
-* Focus on testing behaviour rather than state
-* Commit often, with good commit messages
-* Single Responsibility Principle and encapsulation
-* Clear and readable code
+
```

Comment:
Great use of a model! It's clear, displays different routes and what each route does.